### ISSUE TITLE
unnecessery recreation of unique constraints

### ISSUE THREAD

**Issue by neivkovic** [liquibase/liquibase-hibernate#191]
Hi!

We use in our project liquibase-maven-plugin with the newest version, if we create a diff between our mariadb and the hibernate class files. Every unique constraint is dropped and added again. 

Will be there in the next future improvements?

Here an example of the mentioned drop/create changeset:

```
<changeSet author="nemo (generated)" id="1542210363231-27">
      <dropUniqueConstraint constraintName="UKq61v8b5qvs3cxalvj88b9lcel" tableName="config"/>
      <addUniqueConstraint columnNames="host"
                           constraintName="UKq61v8b5qvs3cxalvj88b9lcel"
                           tableName="config"/>
</changeSet>
```

Our project settings:

```
<artifactId>spring-boot-starter-parent</artifactId>
<groupId>org.springframework.boot</groupId>
<version>2.0.3.RELEASE</version>
<liquibase.version>3.6.2</liquibase.version>
<liquibase-hibernate5.version>3.6</liquibase-hibernate5.version>
```

liquibase-maven-plugin:

```
<groupId>org.liquibase</groupId>
<artifactId>liquibase-maven-plugin</artifactId>
<version>3.6.2</version>
```

```
<groupId>org.liquibase.ext</groupId>
<artifactId>liquibase-hibernate5</artifactId>
<version>3.6</version>
```

Regards
Nenad

┆Issue is synchronized with this [Jira Bug](https://datical.atlassian.net/browse/LB-433) by [Unito](https://www.unito.io/learn-more)

**Comment by benoitheinrich** [comment]
Hey, I've got the exact same problem here and I actually just setup a github project that reproduce this problem.

If someone could have a look at it, thanks :+1: 

https://github.com/benoitheinrich/liquibase-hibernate-bug

**Comment by benoitheinrich** [comment]
Anyone interested by a fix I've published something that so far seems to work.

This is available at the repo indicated above.

This also fixes other issues with that plugin, so enjoy !!

**Comment by benoitheinrich** [comment]
btw this is probably a duplicate of #173

**Comment by timothy-volvo** [comment]
@neivkovic, I ran into a similar issue with Postgres. After some research it appears the database snapshot had the cluster attribute set to false, but the Hibernate snapshot had it set to null. I updated the liquibase-hibernate logic to alway set the cluster attribute to false as I didn't see a way to set it to true from the hibernate configuration. This resolved my issue. I submitted the fix as PR #204.
